### ISSUE TITLE
Tpetra:  compilation error?  configuration change? -- CrsGraph getLocalViewRawConst, getGlobalViewRawConst

### ISSUE THREAD

**Issue by kddevin** [trilinos/Trilinos#1808]
I'm seeing the compilation errors below using a configuration command that has worked in the past.  Has there been a change in the cmake flags needed for builds?  If so, what is now needed?
```
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_SerialWrapperNode.cpp:71:3:   required from here
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:5615:72: error: no matching function for call to 'Tpetra::CrsGraph<int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial> >::getLocalViewRawConst(int*&, int&, const Tpetra::RowInfo&) const'
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:5615:72: note: candidate is:
In file included from Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_SerialWrapperNode.cpp:54:0:
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:1332:3: note: LocalOrdinal Tpetra::CrsGraph<LO, GO, N, isClassic>::getLocalViewRawConst(const LocalOrdinal*&, LocalOrdinal&, const Tpetra::RowInfo&) const [with LocalOrdinal = int; GlobalOrdinal = int; Node = Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial>; bool classic = false]
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:1332:3: note:   no known conversion for argument 1 from 'int*' to 'const int*&'
In file included from Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_SerialWrapperNode.cpp:54:0:
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:5637:73: error: no matching function for call to 'Tpetra::CrsGraph<int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial> >::getGlobalViewRawConst(int*&, int&, const Tpetra::RowInfo&) const'
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:5637:73: note: candidate is:
In file included from Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_SerialWrapperNode.cpp:54:0:
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:1565:3: note: LocalOrdinal Tpetra::CrsGraph<LO, GO, N, isClassic>::getGlobalViewRawConst(const GlobalOrdinal*&, LocalOrdinal&, const Tpetra::RowInfo&) const [with LocalOrdinal = int; GlobalOrdinal = int; Node = Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial>; bool classic = false]
Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:1565:3: note:   no known conversion for argument 1 from 'int*' to 'const int*&'
```

Cmake command:
```
cmake \
-D CMAKE_BUILD_TYPE:STRING="DEBUG" \
-D CMAKE_INSTALL_PREFIX:FILEPATH="/home/kddevin/tmp/tmp/Trilinos/Obj_zoltan2" \
-D CMAKE_VERBOSE_MAKEFILE:BOOL=OFF \
\
-D MPI_BIN_DIR:PATH=/usr/lib64/openmpi/bin \
-D TPL_ENABLE_MPI:BOOL=ON \
-D MPI_EXEC_MAX_NUMPROCS:STRING=11 \
\
-D TPL_ENABLE_BinUtils:BOOL=OFF \
-D TPL_ENABLE_Pthread:BOOL=OFF \
\
-D CMAKE_C_FLAGS:STRING="-fopenmp -g -Wall -Wclobbered  -Wempty-body  -Wignored-qualifiers -Wmissing-field-initializers  -Wmissing-parameter-type -Wold-style-declaration -Woverride-init  -Wsign-compare  -Wtype-limits   -Wuninitialized -Winit-self -fstrict-aliasing " \
-D CMAKE_CXX_FLAGS:STRING="-fopenmp -g -Wall -Wclobbered  -Wempty-body  -Wignored-qualifiers -Wmissing-field-initializers  -Wsign-compare  -Wtype-limits   -Wuninitialized -Winit-self -fstrict-aliasing " \
-D CMAKE_Fortran_FLAGS:STRING="-g " \
\
-D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
-D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
-D Trilinos_ENABLE_TESTS:BOOL=OFF \
-D Trilinos_ENABLE_EXAMPLES:BOOL=OFF \
-D Trilinos_VERBOSE_CONFIGURE:BOOL=OFF \
\
-D Trilinos_ENABLE_Epetra:BOOL=ON \
-D Trilinos_ENABLE_Pamgen:BOOL=ON \
-D Trilinos_ENABLE_Galeri:BOOL=ON \
\
-D Trilinos_ENABLE_Zoltan2:BOOL=ON \
-D Zoltan2_ENABLE_TESTS:BOOL=ON \
-D Zoltan2_ENABLE_EXAMPLES:BOOL=ON \
-D Zoltan2_ENABLE_Experimental:BOOL=ON \
-D Zoltan2_ENABLE_UseZoltanComm:BOOL=ON \
\
-D TPL_ENABLE_PuLP:BOOL=ON \
-D PuLP_LIBRARY_DIRS:FILEPATH="/home/kddevin/code/PuLP/lib" \
-D PuLP_INCLUDE_DIRS:FILEPATH="/home/kddevin/code/PuLP/include" \
\
-D TPL_ENABLE_Scotch:BOOL=ON \
-D Scotch_LIBRARY_DIRS:FILEPATH="/home/kddevin/code/Scotch/scotch_6.0.3/32bit_openmpi/lib" \
-D Scotch_INCLUDE_DIRS:FILEPATH="/home/kddevin/code/Scotch/scotch_6.0.3/32bit_openmpi/include" \
\
-D TPL_ENABLE_ParMETIS:BOOL=ON \
-D ParMETIS_LIBRARY_DIRS:FILEPATH="/home/kddevin/code/ParMETIS/ParMETIS-4.0.3/32bit_openmpi" \
-D ParMETIS_INCLUDE_DIRS:FILEPATH="/home/kddevin/code/ParMETIS/ParMETIS-4.0.3/32bit_openmpi" \
\
-D Teuchos_ENABLE_STACKTRACE=OFF \
..
```
@trilinos/tpetra 

[@jhux2 : edited for formatting]

**Comment by kddevin** [comment]
This problem exists with gcc 4.7.2., but disappears with gcc 4.9.3.  I'm told the minimum gcc is 4.8.4 (despite the fact that trilinos.org says 4.7.2), so I'll close this issue.

**Comment by jwillenbring** [comment]
I found GCC 4.7.2 still referenced in 3 places. One I just updated, the other two I put a note that the minimum required version of Trilinos is now 4.8.4. I didn't change the other two because one was kind of an announcement that at the time was correct, and the other is probably more out of date and needs some attention in general, and simply updating the version in the text seemed to provide false hope that the approach advocated would work. In any case, I hope it is clear from the site now that 4.8.4 is the minimum. Thanks for pointing this out @kddevin.
